http_service/app.py

```diff
@@ -126,12 +128,54 @@ def is_running(model_name, bug_id):
 
     job_status = job.get_status()
     if job_status in ("running", "started", "queued"):
-        print("Bug classification already running", model_name, bug_id)
         return True
 
     return False
 
 
+def get_bugs_last_change_time(bug_ids):
+    query = {"id": bug_ids, "include_fields": ["last_change_time", "id"]}
+    header = {"X-Bugzilla-API-Key": "", "User-Agent": "bugbug"}
+    response = HTTP_CLIENT.get(
+        BUGZILLA_API_URL, params=query, headers=header, verify=True, timeout=30
+    )
+    response.raise_for_status()
+
+    raw_bugs = response.json()
+
+    bugs = {}
+
+    for bug in raw_bugs["bugs"]:
+        bugs[bug["id"]] = bug["last_change_time"]
+
+    return bugs
+
+
+def is_prediction_invalidated(model_name, bug_id, change_time):
+    # First get the saved change time
+    change_key = change_time_key(model_name, bug_id)
+
+    saved_change_time = redis_conn.get(change_key)
+
+    # If we have no last changed time, the bug was not classified yet or the bug was classified by an old worker
+    if not saved_change_time:
+        # We can have a result without a cache time
+        if redis_conn.get(result_key(model_name, bug_id)):
```

Comment:
I wonder if we couldn't avoid getting the result twice from redis (once here and once in get_bug_classification).

Reply:
We would get it twice only if the cache key is not in Redis, which could happens in three cases:
- The bug was not classified yet
- The bug was classified but expired
- The bug was classified with an old version of the code and it won't happen again.

I will see if I can check if I can check only if the redis key exists.